src/rgw/rgw_rados.cc

```diff
@@ -7345,6 +7345,7 @@ int RGWRados::delete_bucket(rgw_bucket& bucket, RGWObjVersionTracker& objv_track
       return r;
 
     string ns;
```

Comment:
I  think it would be better if we fail bucket delete for object in any namespaces. Because all the objects in other namespaces will end up getting leaked as well.  

So we should fail if ent_map contains more than one element.

By the way "ns" is not initialised to any namespace. was that intended ?